test/blas.cpp

```diff
@@ -319,3 +322,246 @@ TEST(MatrixMultiply, RhsBroadcastBatched) {
         }
     }
 }
+
+////////////// Test C API using special/preallocated output arrays //////////////
+
+float alpha = 1.f;
+int num_alpha  = 1;
```

Comment:
I think `num_alpha` & `num_beta` being `1` in the tests is the reason why we didn't catch the problem of non-changing alpha & beta in CPU and OpenCL backend. See my comments in respective backend blas.cpp file.

Reply:
This was intentional because of https://github.com/arrayfire/arrayfire/pull/2481#discussion_r295248645.